components/sup/src/command/start.rs

```diff
@@ -50,109 +50,41 @@
-                    let depot_client = try!(Client::new(&spec.depot_url, PRODUCT, VERSION, None));
-                    let latest_pkg_data = try!(depot_client.show_package(&spec.ident));
-                    let latest_ident: PackageIdent = latest_pkg_data.get_ident().clone().into();
-                    if &latest_ident > package.ident() {
-                        outputln!("Downloading latest version from Depot: {}", latest_ident);
-                        let new_pkg_data = try!(install::start(&mut ui,
-                                                               &spec.depot_url,
-                                                               &latest_ident.to_string(),
-                                                               PRODUCT,
-                                                               VERSION,
-                                                               Path::new(&*FS_ROOT_PATH),
-                                                               &cache_artifact_path(None),
-                                                               false));
-                        package = try!(PackageInstall::load(&new_pkg_data, Some(&*FS_ROOT_PATH)));
-                    } else {
-                        outputln!("Already running latest.");
-                    };
-                }
-            }
-            start_package(package, cfg, spec)
-        }
-        Err(_) => {
-            outputln!("{} is not installed",
-                      Yellow.bold().paint(spec.ident.to_string()));
-            let new_pkg_data = match local_artifact {
-                Some(artifact) => {
-                    try!(install::start(&mut ui,
-                                        &spec.depot_url,
-                                        &artifact,
-                                        PRODUCT,
-                                        VERSION,
-                                        Path::new(&*FS_ROOT_PATH),
-                                        &cache_artifact_path(None),
-                                        false))
-                }
-                None => {
-                    outputln!("Searching for {} in remote {}",
-                              Yellow.bold().paint(spec.ident.to_string()),
-                              &spec.depot_url);
-                    try!(install::start(&mut ui,
-                                        &spec.depot_url,
-                                        &spec.ident.to_string(),
-                                        PRODUCT,
-                                        VERSION,
-                                        Path::new(&*FS_ROOT_PATH),
-                                        &cache_artifact_path(None),
-                                        false))
-                }
-            };
-            let package = try!(PackageInstall::load(&new_pkg_data, Some(&*FS_ROOT_PATH)));
-            start_package(package, cfg, spec)
-        }
+    if let Some(artifact) = local_artifact {
```

Comment:
I am feeling that starting an artifact from a `.hart` might ultimately be a bad idea. We need to do some not-so-elegant things to solve edge cases that it introduces. In design theory if you create a rule, you know the rule is wrong if you need to create "patch rules" especially if you're patching patches.

I think we only ever provided this because we wanted to save somebody some keystrokes by avoiding them to first need to run the install command on their hart before issueing a start command. As we evolve the idea of a ServiceSpec I don't think there's a place for a local artifact in it anymore - essentially what the ServiceSpec has is a "goal" of what to start. That goal can change depending on the service group that the supervisor joins. For example, if you join a service group that's running a different version of the package than you are and the service group is configured in the rolling update strategy then you're going to be upgraded/downgraded to the version of the software everyone else is running.

What do you think? Are we just doing this to save some keystrokes? Should we just ask people to run install first?